src/components/dialog/dialog.interactions.stories.tsx

```diff
@@ -0,0 +1,232 @@
+
+    const candidates = canvas.getAllByRole("button", { name: /open dialog/i });
+    const openButton =
+      candidates.find(
+        (el) => el.getAttribute("data-role") === "dialog-trigger",
+      ) ?? candidates[0];
+
+    await userEvent.click(openButton);
+    await userInteractionPause(300);
+
+    const dialog = portal.getByRole("dialog");
+    const cancelBtn = within(dialog).getByRole("button", { name: /^cancel$/i });
+    await userEvent.click(cancelBtn);
+    await userInteractionPause(300);
+  },
+};
+
+export const FocusManagement: Story = {
+  render: function FocusManagementRender() {
+    const [isOpen, setIsOpen] = useState(false);
+    const buttonRef = useRef<HTMLButtonElement>(null);
+
+    return (
+      <>
+        <Button
+          ref={buttonRef}
+          onClick={() => setIsOpen(true)}
+          data-role="open-dialog-btn"
+        >
+          Open Dialog
+        </Button>
+
+        <Dialog
+          open={isOpen}
+          title="Title"
+          subtitle="Subtitle"
+          onCancel={() => setIsOpen(false)}
+          showCloseIcon
+          focusFirstElement={buttonRef}
+        >
+          <Form
+            stickyFooter
+            leftSideButtons={
+              <Button onClick={() => setIsOpen(false)}>Cancel</Button>
+            }
+            saveButton={
+              <Button buttonType="primary" type="submit">
+                Save
+              </Button>
+            }
+          >
+            <Typography>
+              This is an example of a dialog with a Form as content
+            </Typography>
+            <Textbox label="First Name" />
+            <Textbox label="Middle Name" />
+            <Textbox label="Surname" />
+            <Textbox label="Birth Place" />
+            <Textbox label="Favourite Colour" />
+            <Textbox label="Address" />
+          </Form>
+        </Dialog>
+      </>
+    );
+  },
+
+  play: async ({ canvasElement }) => {
+    if (!allowInteractions()) return;
+
+    const canvas = within(canvasElement);
+    const portal = within(canvasElement.ownerDocument.body);
+
+    const candidates = canvas.getAllByRole("button", { name: /open dialog/i });
+    const openButton =
+      candidates.find(
+        (el) => el.getAttribute("data-role") === "open-dialog-btn",
+      ) ?? candidates[0];
+
+    await userEvent.click(openButton);
+    await userInteractionPause(300);
+
+    const dialog = portal.getByRole("dialog");
+
+    await userEvent.tab();
+    await userInteractionPause(50);
+
+    await userEvent.tab();
+    await userInteractionPause(50);
+
+    const firstInput = within(dialog).getByLabelText(/first name/i);
+    await userEvent.type(firstInput, "Jane Doe", { delay: 30 });
+    await userInteractionPause(100);
+
+    await userEvent.tab();
+    await userInteractionPause(50);
+    await userEvent.tab();
+    await userInteractionPause(50);
+
+    const closeIcon = within(dialog).getByRole("button", { name: /close/i });
```

Comment:
comment: if we click the close icon the snapshot will just be the open button. There's limited value in interactions with Dialog as we can control the open state but capturing the focus state is worth it imo